share/spice/thesaurus/thesaurus.js

```diff
@@ -2,83 +2,90 @@
     env.ddg_spice_thesaurus = function(api_result) {
         "use strict";
 
-        if (!api_result){
+        if (!api_result || api_result.error) {
             return Spice.failed("thesaurus");
         }
 
-        // Get the query and the mode (trigger words)
-        // The mode tells us what to return
-        // e.g. you want the antonym but not the synonym
-        var script = $('[src*="/js/spice/thesaurus/"]')[0],
-            source = $(script).attr("src"),
-            match  = source.match(/\/js\/spice\/thesaurus\/([^\/]+)\/([^\/]+)/),
-            query  = match[1],
-            mode   = match[2];
+        // Retrieve the word we did a lookup on
+        var script = $('[src*="/js/spice/thesaurus/"]')[0];
+        var source = $(script).attr("src");
+        var query = source.match(/thesaurus\/([^\/]+)/)[1];
+        query = decodeURIComponent(query);
+        
+        // Search the results for the word type (noun, adjective etc) with the most entries
+        var wordTypeWithMostEntries;
+        var largestWordTypeEntryTotal = 0;
+        
+        Object.keys(api_result).forEach(function(wordType) { // wordType = ajective, noun etc...
```

Comment:
If you want, we have jQuery available, so we typically use `$.each()`. I don't think I've seen anyone use `Object.keys()` before :smile: